server/controller/src/main/java/ai/starwhale/mlops/schedule/SwTaskSchedulerImpl.java

```diff
@@ -67,27 +67,30 @@ public void schedule(Task task) {
                 log.info("task {} is canceled, skip schedule", task.getId());
                 return;
             }
-            ContainerSpecification containerSpecification = taskContainerSpecificationFinder.findCs(task);
-            RunSpec runSpec = RunSpec.builder()
-                    .image(containerSpecification.getImage())
-                    .envs(containerSpecification.getContainerEnvs())
-                    .command(containerSpecification.getCmd())
-                    .resourcePool(task.getStep().getResourcePool())
-                    .requestedResources(task.getTaskRequest().getRuntimeResources())
-                    .build();
             RunEntity runEntity = RunEntity.builder()
                     .taskId(task.getId())
-                    .runSpec(runSpec)
                     .logDir(task.getResultRootPath().logDir())
                     .build();
             runMapper.insert(runEntity);
             Run run = Run.builder()
                     .id(runEntity.getId())
                     .taskId(task.getId())
-                    .runSpec(runSpec)
                     .logDir(runEntity.getLogDir())
                     .build();
             task.setCurrentRun(run);
+
+            ContainerSpecification containerSpecification = taskContainerSpecificationFinder.findCs(task);
+            RunSpec runSpec = RunSpec.builder()
+                    .image(containerSpecification.getImage())
+                    .envs(containerSpecification.getContainerEnvs())
+                    .command(containerSpecification.getCmd())
+                    .resourcePool(task.getStep().getResourcePool())
+                    .requestedResources(task.getTaskRequest().getRuntimeResources())
+                    .build();
+
+            runMapper.updateSpec(runEntity.getId(), runSpec);
```

Comment:
just modify the two lines below is enough, I think.

https://github.com/star-whale/starwhale/blob/02fd459fac2491777889fccb054e2f79bbe2deeb/server/controller/src/main/java/ai/starwhale/mlops/schedule/impl/k8s/RunExecutorK8s.java#L74
https://github.com/star-whale/starwhale/blob/02fd459fac2491777889fccb054e2f79bbe2deeb/server/controller/src/main/java/ai/starwhale/mlops/schedule/impl/docker/RunExecutorDockerImpl.java#L133